applications/asset_tracker_v2/src/modules/data_module.c

```diff
@@ -983,13 +983,24 @@ static void agps_request_handle(struct nrf_modem_gnss_agps_data_frame *incoming_
 	 * to send the request.
 	 */
 #if defined(CONFIG_NRF_CLOUD_AGPS) && defined(CONFIG_NRF_CLOUD_MQTT)
+	uint32_t sv_mask_ephe = incoming_request->sv_mask_ephe;
+
+#if defined(CONFIG_NRF_CLOUD_PGPS)
+	incoming_request->sv_mask_ephe = 0;
+#endif
+
 	err = nrf_cloud_agps_request(incoming_request);
+	incoming_request->sv_mask_ephe = sv_mask_ephe; /* restore if changed for P-GPS */
```

Comment:
Is it necessary to restore the ephemeris mask here? It doesn't seem to be used anywhere post `nrf_cloud_agps_request`

Reply:
It's passed in by reference to `agps_request_handle()`, so it seems bad form to modify it.

But, I thought of another way to do this that doesn't require modifying it.